lib/ansible/modules/crypto/openssl_pkcs12.py

```diff
@@ -237,7 +238,48 @@ def _check_pkey_passphrase():
         if not state_and_perms:
             return state_and_perms
 
-        return _check_pkey_passphrase
+        if os.path.exists(self.path) and module.params['action'] == 'export':
+            dummy = self.generate(module)
+            self.src = self.path
+            try:
+                pkcs12_privatekey, pkcs12_certificate, pkcs12_other_certificates, pkcs12_friendly_name = self.parse()
+            except crypto.Error:
+                return False
+            if (pkcs12_privatekey is not None) and (self.privatekey_path is not None):
+                expected_pkey = crypto.dump_privatekey(crypto.FILETYPE_PEM,
+                                                       self.pkcs12.get_privatekey())
+                if pkcs12_privatekey != expected_pkey:
+                    return False
+            elif bool(pkcs12_privatekey) != bool(self.privatekey_path):
+                return False
+
+            if (pkcs12_certificate is not None) and (self.certificate_path is not None):
+
+                expected_cert = crypto.dump_certificate(crypto.FILETYPE_PEM,
+                                                        self.pkcs12.get_certificate())
+                if pkcs12_certificate != expected_cert:
+                    return False
+            elif bool(pkcs12_certificate) != bool(self.certificate_path):
+                return False
+
+            if (pkcs12_other_certificates is not None) and (self.ca_certificates is not None):
+                expected_ca_certs = [crypto.dump_certificate(crypto.FILETYPE_PEM,
+                                                             ca_cert) for ca_cert in self.pkcs12.get_ca_certificates()]
+                if set(pkcs12_other_certificates) != set(expected_ca_certs):
+                    return False
+            elif bool(pkcs12_other_certificates) != bool(self.ca_certificates):
+                return False
+
+            if pkcs12_privatekey:
```

Comment:
Maybe add a comment that OpenSSL only loads the friendly name if a private key is in the PKCS12 file.